### ISSUE TITLE
Documentation should indicate which Mobjects are non-deterministic

### ISSUE THREAD

**Issue by Darylgolden** [ManimCommunity/manim#1408]
## Enhancement proposal
#1283 made it such that Mobjects that rely call on the random function in some way are no longer deterministic. This should be properly documented to avoid confusion. The non-deterministic behavior of some Mobjects, like `Graph`, are not obvious at all.

**Comment by joybh98** [comment]
@Darylgolden can I work on this issue ?

**Comment by Darylgolden** [comment]
> 
> 
> @Darylgolden can I work on this issue ?

Sure! Assigned you.

**Comment by joybh98** [comment]
@Darylgolden could you give me some advice on how to find which mobjects are deterministic as I'm new to this project!

**Comment by Darylgolden** [comment]
> 
> 
> @Darylgolden could you give me some advice on how to find which mobjects are deterministic as I'm new to this project!

Unfortunately I don't think there's an easy way to find out. I started a discussion in #manim-dev in the Discord to see if people can think of more non-deterministic Mobjects.

**Comment by PhotonSpheres** [comment]
> @Darylgolden could you give me some advice on how to find which mobjects are deterministic as I'm new to this project!

Thanks for your help! These are not Mobjects but functions that rely on some randomness:

- `random_color()`
- `random_bright_color()`

They are in the `manim.utils.color` module. A seed for those functions would be great (I think @kolibril13 suggested that before?).

**Comment by kolibril13** [comment]
> I think @kolibril13 suggested that before?

Yes, I see you found the issue.

**Comment by Darylgolden** [comment]
> They are in the `manim.utils.color` module. A seed for those functions would be great (I think @kolibril13 suggested that before?).

True, but that's not really relevant to this specific issue (which is about documentation).

> @Darylgolden could you give me some advice on how to find which mobjects are deterministic as I'm new to this project!

Perhaps searching the repository for "random" might reveal some things we've forgotten to mention here.

**Comment by joybh98** [comment]
> True, but that's not really relevant to this specific issue (which is about documentation).

Thanks @Darylgolden I'll join the discord and check for any updates !

**Comment by behackl** [comment]
There are only very few select methods that rely on randomness at this point, and for most of them this should be implied by their name or nature (e.g., `Mobject.shuffle` or the `"random"` layout for Graphs).

I think there is nothing to do here currently.
